### PR TITLE
java: add proper lint task to CI

### PR DESCRIPTION
**Pull request by axic**
Fixes https://github.com/ethereum/evmc/pull/559#discussion_r516536807

### CODE REVIEW

**Comment by axic** [comment]
Current failures:
```
* What went wrong:
Execution failed for task ':spotlessJava'.
> The following files had format violations:
      java/src/main/java/org/ethereum/evmc/EvmcLoaderException.java
          @@ -4,7 +4,7 @@
           package·org.ethereum.evmc;
           
           public·class·EvmcLoaderException·extends·Exception·{
          -···public·EvmcLoaderException(String·message)·{
          -······super(message);
          -···}
          +··public·EvmcLoaderException(String·message)·{
          +····super(message);
          +··}
           }
      java/src/main/java/org/ethereum/evmc/EvmcVm.java
          @@ -3,8 +3,6 @@
           //·Licensed·under·the·Apache·License,·Version·2.0.
           package·org.ethereum.evmc;
           
          -import·org.ethereum.evmc.EvmcLoaderException;
          -
           import·java.nio.ByteBuffer;
           import·java.util.Objects;
           
          @@ -21,7 +19,7 @@
           ···*·This·method·loads·the·specified·evm·shared·library·and·loads/initializes·the·jni·bindings.
           ···*
           ···*·@param·filename·/path/filename·of·the·evm·shared·object
          -···*·@throws·EvmcLoaderException
          +···*·@throws·org.ethereum.evmc.EvmcLoaderException
           ···*/
           ··public·static·EvmcVm·create(String·filename)·throws·EvmcLoaderException·{
           ····if·(!EvmcVm.isEvmcLibraryLoaded)·{
          @@ -49,7 +47,7 @@
           ···*
           ···*·@param·Path·to·the·dynamic·object·representing·the·EVM·implementation.
           ···*·@return·Internal·object·pointer.
          -···*·@throws·EvmcLoaderException
          +···*·@throws·org.ethereum.evmc.EvmcLoaderException
           ···*/
           ··private·static·native·ByteBuffer·load_and_create(String·filename)·throws·EvmcLoaderException;
           
          @@ -152,9 +150,7 @@
           ··/**·get·size·of·result·struct·*/
           ··private·static·native·int·get_result_size();
           
          -··/**
          -···*·This·method·cleans·up·resources.
          -···*/
          +··/**·This·method·cleans·up·resources.·*/
           ··@Override
           ··public·void·close()·{
      ... (1 more lines that didn't fit)
  Violations also present in:
      java/src/main/java/org/ethereum/evmc/Host.java
  Run 'gradlew spotlessApply' to fix these violations.
```

File: bindings/java/Makefile [hunk 1]
```diff
@@ -8,6 +8,9 @@ SOURCE_DIR = $(realpath $(CURDIR)/../..)
 OUT_DIR = $(CURDIR)/c/build
 BUILD_DIR = $(OUT_DIR)/_cmake_build
 
+lint: gradlew
+	./gradlew --no-daemon clean spotlessCheck
+
 build: gradlew $(OUT_DIR)/lib/libevmc-java.so
```
Comment (axic) [inline bindings/java/Makefile:14]: Could also make build depend on lint. And rename the gradlew task to gradle-setup. And reorder the file.

File: circle.yml [hunk 2]
```diff
@@ -301,14 +301,15 @@ jobs:
...
       - run:
           name: "Java Build"
-          command: |
-            cd bindings/java && make clean build
+          command: cd bindings/java && make clean build
       - run:
           name: "Java Test"
-          command: |
-            cd bindings/java && make test
+          command: cd bindings/java && make test
```
Comment (axic) [inline circle.yml:312]: Can remove these whitespaces changes.

**Review by atoulme — APPROVED** [review]
lgtm

**Review by chfast — APPROVED** [review]

**Pull request merged**